### ISSUE TITLE
Zend\Cache\Service\StorageCacheFactory: Order of Redis adapter options may cause Exception

### ISSUE THREAD

**Issue by eiwaen** [zendframework/zf2#6381]
Depending on the order of the adapter options, the following Exception may be thrown:
`Zend\Cache\Exception\InvalidArgumentException' with message 'Missing required server host'`

This **works**:
```php
'caches' => array(
    'redis' => array(
        'adapter' => array(
            'name' => 'Zend\Cache\Storage\Adapter\Redis',
            'options' => array(
                'resource_id' => 'myresource',
                'server' => 'hostname',
                'password' => 'password',
            ),
        ),
    ),
),
```

This **doesn't work**:
```php
'caches' => array(
    'redis' => array(
        'adapter' => array(
            'name' => 'Zend\Cache\Storage\Adapter\Redis',
            'options' => array(
                'server' => 'hostname', // Defined under resource "default"
                'resource_id' => 'myresource',
                'password' => 'password', // Would be defined under resource "myresource", but exception is thrown (missing hostname when setting password)
            ),
        ),
    ),
),
```

**Comment by lucian303** [comment]
This is obviously a bug. What are we supposed to do guess what order an unordered array of options should go in? At the very least, add a big note to the documentation so that people know the order matters and don't waste hours trying to figure out what's wrong.

**Comment by marc-mabe** [comment]
@eiwaen @lucian303 Should be "fixed" by https://github.com/zendframework/zend-cache/issues/91
Please test it

**Comment by marc-mabe** [comment]
@Ocramius @weierophinney This one is fixed by zendframework/zend-cache#91 and can be closed
